tests/Tests.Json.Decode.fs

```diff
@@ -1064,11 +1083,12 @@ Expecting an object but instead got:
 
                 equal expectedMissingField actualMissingField
 
-                let expectedUndefinedField = Ok(None)
-                let actualUndefinedField =
-                    Decode.fromString (Decode.optional "something_undefined" Decode.string) json
+                // TODO: Should this test pass? The field is present but not a string
```

Comment:
Hum, if the value is `null` I would consider it ok and generate `None`.

The reason being, I don't think other JSON libraries don't output `None` fields. And also, our own Encode is generating `null` if `None` is pass to it. [Source](https://github.com/thoth-org/Thoth.Json/blob/master/src/Encode.fs#L327-L328)

However, if the field is present but with an unexpected Type then I would generate an error as asked by #3.

But I need to check this [see the comment at this line](https://github.com/thoth-org/Thoth.Json/pull/2/commits/1e668a8a77b19352bf28952b0ac039c85428f937#diff-0085955556097501404f68ae2f0b3fcdR1146)